smarts/env/hiway_env.py

```diff
@@ -129,22 +129,56 @@ def __init__(
         if visdom:
             visdom_client = VisdomClient()
 
-        self._smarts = SMARTS(
-            agent_interfaces=agent_interfaces,
-            traffic_sim=SumoTrafficSimulation(
+        all_sumo = HiWayEnv.check_scenario_versions(scenarios)
+        traffic_sim = None
+        if not all_sumo:
+            # We currently only support the Native SUMO Traffic Provider and Social Agents for SUMO maps
+            if zoo_addrs:
+                warnings.warn("`zoo_addrs` can only be used with SUMO scenarios")
+                zoo_addrs = None
+            warnings.warn(
+                "We currently only support the Native SUMO Traffic Provider and Social Agents for SUMO maps."
+                "All scenarios passed need to be of SUMO, to enable SUMO Traffic Simulation and Social Agents."
+            )
+            pass
+        else:
+            from smarts.core.sumo_traffic_simulation import SumoTrafficSimulation
+
+            traffic_sim = SumoTrafficSimulation(
                 headless=sumo_headless,
                 time_resolution=fixed_timestep_sec,
                 num_external_sumo_clients=num_external_sumo_clients,
                 sumo_port=sumo_port,
                 auto_start=sumo_auto_start,
                 endless_traffic=endless_traffic,
-            ),
+            )
+            zoo_addrs = zoo_addrs
+
+        self._smarts = SMARTS(
+            agent_interfaces=agent_interfaces,
+            traffic_sim=traffic_sim,
             envision=envision_client,
             visdom=visdom_client,
             fixed_timestep_sec=fixed_timestep_sec,
             zoo_addrs=zoo_addrs,
         )
 
+    @staticmethod
+    def check_scenario_versions(scenarios):
+        num_sumo = 0
+        scenario_list = Scenario.get_scenario_list(scenarios)
+        for scenario_root in scenario_list:
+            try:
+                road_map, _ = Scenario.build_map(scenario_root)
+            except FileNotFoundError:
+                raise FileNotFoundError(
+                    f"Unable to find network file in map_source={scenario_root}."
+                )
+            if road_map.source.endswith(".net.xml"):
```

Comment:
I think this should instead be:
```python
if isinstance(road_map, SumoRoadNetwork):
```

Reply:
Then i would have to import `SumoRoadNetwork` and that would be an unwanted dependency on `OpenDriveRoadNetwork`.

Reply:
Well you would only have to import it here (in this function), and nothing in the core imports this file anyway.  So there's no cross-dependency from OpenDrive to Sumo.

You already anyway have a sort of "dependency" implicit in your assumption of what the filename extension for Sumo will be.  This breaks encapsulation (only `SumoRoadNetwork` and the map builder should determine the filenames and extensions from which such an object can be made) -- i.e., if Sumo tomorrow changed their file naming conventions, you would have to change the code here too if you leave it as is.  

(btw, the stuff in sstudio is a bad exception, and is something I want to change in a future release... but here we don't need to repeat that mistake.)

Reply:
```
Well you would only have to import it here (in this function), and nothing in the core imports this file anyway. So there's no cross-dependency from OpenDrive to Sumo.
```
I am not sure what you meant here. Any `OpenDRIVE` scenario running on this environment will call this method to check if none of the scenario passed are SUMO. And if import `SumoRoadNetwork` here wouldn't that create a dependency on OpenDRIVE since importing `SumoRoadNetwork` will also import the SUMO import dependencies of `sumo_road_network.py`?

Reply:
But what I meant above is that this method isn't called from anywhere else but within this code -- indeed, it should really be renamed to `_check_scenario_versions()` -- it's not called from `opendrive_road_network.py`, for example.

It seems you want to "have your cake here and eat it too".   You really _do_ have a dependency on Sumo in this environment because it needs to determine whether traffic is supported and that's only the case for scenarios with Sumo maps (but see below).  The _only_ _clean_ way you can determine whether you have a Sumo map is to use the map builder_fn to create the map for the scenario and then see what you get.  (This _cannot_ be done merely by looking at the filename alone, no matter what is done in the _default_ map builder; custom map builders can do whatever they want and support whatever file naming conventions they want.)

So yes, anything that uses this environment will have a dependency on `SumoRoadNetwork` for now -- but even if you don't import it, the dependency is still there, it's just implicit (and breaking encapsulation) instead.

In the future, we can implement a better approach.  A simple first step would be to add a method something like `def supports_traffic_simulations() -> bool` to the `RoadMap` interface, at which point we can call that here instead of doing the "`isinstance()`" that I'm suggesting.  Something like this _would_ allow you to have your cake and eat it too.  But I don't think we want to do that now because we don't yet know how an abstract traffic simulation interface will interact with the `RoadMap` interface.  (For example, we might want to support custom builders for that too, and if so, would use that  instead of using the `RoadMap` interface here.)